### PR TITLE
Update to experimental plugin 0.4.0 and graddle 2.8

### PR DESCRIPTION
**Pull request by Whiteseeker**
I just updated the experimental plugin to 0.4.0 and graddle to 2.8 (until the instant run is fixe in 0.6.0-alphaX)
And clean useless code * empty lines

### CODE REVIEW

**Comment by ggfan** [comment]
I think instant-run is Android Studio feature, need to be enable/disabled in Android Studio; gradle-experimental plugin has conflict with it, does not matter which plugin version ( 0.4.0 or 0.6.0) it is. As along as using android studio 2.0 preview and above, you will have to disable "instant -run" in android studio ( not gradle plugin) so gradle-experimental plugin could work.

May you correct/confirm my understanding, if it is true, directly upgrade to 0.6.0?

**Comment by Whiteseeker** [comment]
Instant run conflict with gradle 2.9 (which is necessary for experimental
plugin 0.6.0-alpha2) from my tests. That why I updated only to 0.4.0.

**Comment by ggfan** [comment]
yes, confirmed: instant run needs gradle 2.9; gradle-experimental 0.6.x needs 2.9, but also conflicts with instant run.  you are correct, could only upgrade to 0.4.0

File: howie/build.gradle [hunk 1]
```diff
@@ -29,29 +29,29 @@ model {
         }
     }
 
+    compileOptions.with {
```
Comment (ggfan) [inline howie/build.gradle]: may you check that: this should be inside the android { } scope?  thanks

File: howie/build.gradle [hunk 2]
```diff
@@ -29,29 +29,29 @@ model {
...
-        cppFlags += "-std=c++11"
+        moduleName = "howie"
+        cppFlags.add("-std=c++11")
 // Uncomment this line to get more debug spew.
-//        cppFlags += "-DHOWIE_TRACE_LEVEL=1"
-        ldLibs += "OpenSLES"
-        ldLibs += "log"
+//        cppFlags.add("-DHOWIE_TRACE_LEVEL=1")
+        ldLibs.add("OpenSLES")
+        ldLibs.add("log")
```
Comment (ggfan) [inline howie/build.gradle]: would this format ldLibs.addAll(["...", "..."]) ok?

File: samples/HelloLowLatencyIO/build.gradle [hunk 3]
```diff
@@ -30,30 +34,23 @@ model {
...
 
     android.ndk {
-        moduleName += "hello_low_latency_io"
-        ldLibs += "OpenSLES"
-        ldLibs += "log"
-        CFlags += "-I${howieIncludePath}".toString()
-        cppFlags += "-I${howieIncludePath}".toString()
-        cppFlags += "-std=c++11"
+        moduleName = "hello_low_latency_io"
+        ldLibs.add("OpenSLES")
```
Comment (ggfan) [inline samples/HelloLowLatencyIO/build.gradle]: use arrays to save some typings [ probably more typing for you, :-)]?

File: samples/HelloLowLatencyOutput/build.gradle [hunk 4]
```diff
@@ -17,6 +17,11 @@ model {
         }
     }
 
+    compileOptions.with {
```
Comment (ggfan) [inline samples/HelloLowLatencyOutput/build.gradle]: same as before, inside android{}?

File: samples/HelloLowLatencyOutput/build.gradle [hunk 5]
```diff
@@ -28,22 +33,19 @@ model {
...
 
     android.ndk {
-        moduleName += "hello_low_latency_output"
-        ldLibs += "OpenSLES"
-        ldLibs += "log"
-        CFlags += "-I${howieIncludePath}".toString()
-        cppFlags += "-I${howieIncludePath}".toString()
-        cppFlags += "-std=c++11"
+        moduleName = "hello_low_latency_output"
+        ldLibs.add("OpenSLES")
```
Comment (ggfan) [inline samples/HelloLowLatencyOutput/build.gradle]: same as before: use arrays?

File: samples/HelloLowLatencyOutput/build.gradle [hunk 6]
```diff
@@ -28,22 +33,19 @@ model {
...
+        ldLibs.add("log")
+        CFlags.add("-I${howieIncludePath}".toString())
+        cppFlags.add("-I${howieIncludePath}".toString())
```
Comment (ggfan) [inline samples/HelloLowLatencyOutput/build.gradle]: arrays? like
 cppFlags.addAll(["flag1",
                             "flag2" ])

File: howie/build.gradle (hunk 1 above)
Comment (Whiteseeker) [inline howie/build.gradle]: Not in 0.4.0, but in 0.6.0-alpha2 yes

File: samples/HelloLowLatencyOutput/build.gradle (hunk 4 above)
Comment (Whiteseeker) [inline samples/HelloLowLatencyOutput/build.gradle]: Not in 0.4.0, but in 0.6.0-alpha2 yes

File: samples/HelloLowLatencyOutput/build.gradle (hunk 6 above)
Comment (Whiteseeker) [inline samples/HelloLowLatencyOutput/build.gradle]: Yep, will fix all the array (^_^)

**Pull request merged**